main.go

```diff
@@ -49,23 +57,39 @@ func main() {
 	}
 
 	portClient, err := cli.New(portBaseURL,
-		cli.WithHeader("User-Agent", fmt.Sprintf("port-k8s-exporter/0.1 (statekey/%s)", exporterConfig.StateKey)),
 		cli.WithClientID(portClientId), cli.WithClientSecret(portClientSecret),
 		cli.WithDeleteDependents(deleteDependents), cli.WithCreateMissingRelatedEntities(createMissingRelatedEntities),
+		cli.WithHeader("User-Agent", fmt.Sprintf("port-k8s-exporter/0.1 (statekey/%s)", stateKey)),
 	)
 
 	if err != nil {
 		klog.Fatalf("Error building Port client: %s", err.Error())
 	}
 
-	err = integration.NewIntegration(portClient, stateKey)
+	exporterConfig, err := config.New(configFilePath, resyncInterval, stateKey, eventListenerType)
 	if err != nil {
-		klog.Fatalf("Error creating K8s integration: %s", err.Error())
+		klog.Fatalf("Error building Port K8s Exporter config: %s", err.Error())
 	}
 
+	_, err = integration.GetIntegrationConfig(portClient, stateKey)
+	if err != nil {
+		err = integration.NewIntegration(portClient, stateKey, exporterConfig)
+		if err != nil {
+			klog.Fatalf("Error creating K8s integration: %s", err.Error())
+		}
+	}
+	cli.WithHeader("User-Agent", fmt.Sprintf("port-k8s-exporter/0.1 (statekey/%s)", exporterConfig.StateKey))(portClient)
+
 	klog.Info("Starting controllers handler")
-	controllersHandler := handlers.NewControllersHandler(exporterConfig, k8sClient, portClient)
-	controllersHandler.Handle(stopCh)
+	handler, _ := InitiateHandler(exporterConfig, k8sClient, portClient)
+	eventListener := event_listener.NewEventListener(stateKey, eventListenerType, handler, portClient)
+	err = eventListener.Start(func(handler *handlers.ControllersHandler) (*handlers.ControllersHandler, error) {
+		handler.Stop()
+		return InitiateHandler(exporterConfig, k8sClient, portClient)
+	})
+	if err != nil {
+		klog.Fatalf("Error starting event listener: %s", err.Error())
```

Comment:
If the evnt listener fails it means i can update the configuration

for any ocean integration it means death

the client should handle any issue of the event listener because its one of the core functionality